packages/bot-strategy-runner/test/ConfigBuilder.ts

```diff
@@ -0,0 +1,156 @@
+const minimalConfig: strategyRunnerConfig = {
+  botNetwork: "mainnet_mnemonic",
+  globalAddressWhitelistUrls: [
+    "https://raw.githubusercontent.com/UMAprotocol/protocol/4612097ead953b89daa6e237cdb6c704460025dd/packages/affiliates/payouts/devmining-status.json",
+  ],
+  liquidatorSettings: { enableBotType: true },
+  disputerSettings: { enableBotType: true },
+  monitorSettings: { enableBotType: true },
+};
+
+const sampleEMPAddressNotOnWhitelist = "0xAb5801a7D398351b8bE11C439e05C5B3259aeC9B";
+
+// whitelist sample taken from the globalAddress whitelist.
+const sampleWhitelist = [
+  "0xCef85b352CCD7a446d94AEeeA02dD11622289954",
+  "0x56BaBEcb3dCaC063697fE38AB745c10181c56fA6",
+  "0x10E018C01792705BefB7A757628C2947E38B9426",
+  "0xb40BA94747c59d076B3c189E3A031547492013da",
+  "0x46f5E363e69798a74c8422BFb9EDB63e3FB0f08a",
+];
+
+const minimumRequiredConfigKeys = [
+  "botType",
+  "syntheticSymbol",
+  "botIdentifier",
+  "botNetwork",
+  "financialContractAddress",
+  "pollingDelay",
+];
+describe("buildGlobalWhitelist", async function () {
+  it("Correctly fetches global whitelist", async function () {
+    this.timeout(60000);
+    // Construct a whitelist from all current EMPs on the affiliates payout. Append a sample address to the list.
+    const constructedWhitelist = await buildGlobalWhitelist({
+      ...minimalConfig,
+      globalAddressWhitelist: [sampleEMPAddressNotOnWhitelist],
+    });
+    // First address should be the sample and the list returned should contain other valid addresses.
+    assert.equal(constructedWhitelist[0], sampleEMPAddressNotOnWhitelist);
+    assert.isTrue(constructedWhitelist.length > 0);
+    constructedWhitelist.forEach((address: string) => {
+      assert.isTrue(isAddress(address));
+    });
+  });
+  it("Correctly removes black listed addresses", async function () {
+    this.timeout(60000);
+    // Construct a whitelist from all current EMPs on the affiliates payout and remove the first address from the sampleWhitelist.
+    const constructedWhitelist = await buildGlobalWhitelist({
+      ...minimalConfig,
+      globalAddressBlacklist: [sampleWhitelist[0]],
+    });
+    // The returned whitelist should contain additional addresses over and above the sample whitelist - the black listed.
+    assert.isTrue(constructedWhitelist.length > sampleWhitelist.length - 1);
+
+    // Each address in the sampleWhitelist (except for the first one) should be included in the constructed config.
+    sampleWhitelist.splice(1, sampleWhitelist.length).forEach((address: string) => {
+      assert.isTrue(constructedWhitelist.includes(address)); // all addresses, except for the blacklisted address, should be included.
+    });
+  });
+  it("Correctly errors on incorrectly structured external whitelist file", async function () {
+    this.timeout(60000);
+    let errorsThrown = 0; // count the number of errors. note that // assert.throws(()... syntax did not work for some reason.
```

Comment:
i tried this in chai, i think its because these functions are async, did not see a way to do this with async functions